Makefile

```diff
@@ -1,10 +1,65 @@
-test:
+# COLORS
+RED=\033[0;31m
+GREEN=\033[0;32m
+ORANGE=\033[0;33m
+BLUE=\033[0;34m
+PURPLE=\033[0;35m
+CYAN=\033[0;36m
+LIGHTGRAY=\033[0;37m
+DARKGRAY=\033[1;30m
+YELLOW=\033[0;33m
+NC=\033[0m # No Color
+
+chip?=F4
+BSPConfig?=off
+
+# Used to select the MCU (default is STM32F413)
+ifeq ($(chip), F4)
+    MCU_TARGET = STM32F413
+else ifeq ($(chip), L4)
+    MCU_TARGET = STM32L431
+else
+    $(error ${RED}Invalid MCU specified ${NC}(L4 and F4 are the only options))
+endif
+
+# to do: this is hella jank and only here since BSP.C is built for the F4 MCU. 
+# Keeping it optional to use BSP.C is nice, but there are cleaner ways we can do it.
+
+# If BSPConfig = on, we compile BSP.C
+ifeq ($(BSPConfig), on)
+    BSPConfig = on
+endif
+
+bsp_test:
 ifdef test
-	$(MAKE) -C BSP -C STM32F413 -j TARGET=bsp PROJECT_DIR=../.. BUILD_DIR=../../Objects TEST=../Tests/$(test).c
+	$(MAKE) -C BSP -C $(MCU_TARGET) -j TARGET=bsp PROJECT_DIR=../.. BUILD_DIR=../../Objects TEST=../Tests/$(test).c
```

Comment:
Just making sure; it's intended behavior to create a Objects directory under Embedded-Sharepoint, and for PROJECT_DIR to be Embedded-Sharepoint? What's the plan for building when this is integrated as a submodule with the higher level repositories?

Reply:
Yeah, this Makefile shouldn't be called during when it's used as a submodule, instead they'd call the STM32 makefile directly. With this you can just pass in new directories for your project and it'll compile relative to those directories.